markbook.py

```diff
@@ -6,47 +6,48 @@
 
 
 def create_assignment(name: str, due: str, points: int) -> Dict:
-    """Creates an assignment represented as a dictionary
-    
-    Args:
-        name: the name of the assignment.
-        due: the due date for the assignment.
-        points: what the assignment is out of (denominator).
-    Returns:
-        Assignment as a dictionary.
-    """
-    return {}
+  assignment = {
+    "name": name, 
+    "due": due,
+    "points": points
+  }
+  return assignment
 
 
 def create_classroom(course_code: str, course_name: str, period: int, teacher: str) -> Dict:
-    """Creates a classroom dictionary"""
-    return {}
+  classroom = {
+    "course_code": course_code,
+    "course_name": course_name, 
+    "period": period,
+    "teacher": teacher,
+    "student_list": [],
+    "assignment_list": []
+  }
+  return classroom
 
 
-def calculate_average_mark(student: Dict) -> float:
-    """Calculates the average mark of a student"""
-    return 0
+def test_calculate_average_mark(student: Dict) -> float:
+  total = 0
+  counter = 0
+  for i in student['marks']:
```

Comment:
"for every 'index' in the list of marks...."
*Reaches into a bag full of marks, and pulls out.. an index?*

Perhaps:
"for every mark in the list of marks..."

```python
for mark in student["marks"]:
```